llumnix/backends/vllm/scheduler.py

```diff
@@ -110,28 +126,63 @@ def pop_migrating_out_requests_last_stage(self) -> List[SequenceGroupLlumnix]:
         self.migrating_out_request_last_stage.clear()
         return migrating_out_request_last_stage
 
-    def pre_alloc(self, request_id: str, block_num: int) -> List[int]:
+    def pre_alloc(self,
+                  request_id: str,
+                  request_status: RequestStatus,
+                  request_arrival_time: float,
+                  block_num: int) -> List[int]:
+        # Only migrate waiting request when the waiting request is the earliest arrival one
+        # among the requests of dst instance's waiting queue.
+        if request_status == RequestStatus.WAITING_MIGRATING:
+            if (self.waiting and request_arrival_time > self.waiting[0].arrival_time) \
+                or block_num * self.cache_config.block_size > self.prompt_limit:
+                return []
```

Comment:
Can you return None?

Reply:
OK.

Reply:
I'm the only non-bot commenter

Reply:
`return []` is also a common coding style, vllm also applies the `return []` way, so i think we can stay unchanged here.